### PR TITLE
Fix issue #182 where a visible cell won't get reconfigured

### PR DESCRIPTION
**Pull request by nataliq-pp**
https://github.com/iZettle/Form/issues/182

### CODE REVIEW

File: Form/TableKit.swift [hunk 1]
```diff
@@ -363,35 +363,60 @@ extension TableKit: TableAnimatable {
...
+
         view.animate(changes: changes, animation: animation)
 
-        var hasReconfiguredCells = false
-        for indexPath in view.indexPathsForVisibleRows ?? [] {
-            guard let tableIndex = TableIndex(indexPath, in: self.table) else { continue }
-            let row = table[tableIndex]
-            guard let index = from.firstIndex(where: { rowIdentifier(row) == rowIdentifier($0) }) else { continue }
+        if reconfigureVisibleRowsIfNeeded() {
+            // To to refresh cells where the cell height has changed.
```
Comment (enhorn) [inline Form/TableKit.swift:372]: NIT
```suggestion
            // To refresh cells where the cell height has changed.
```

File: Form/TableKit.swift [hunk 2]
```diff
@@ -363,35 +363,60 @@ extension TableKit: TableAnimatable {
...
+            var handledUpdates: [TableIndex] = []
+            for indexPath in view.indexPathsForVisibleRows ?? [] {
+                for change in changes {
+                    if case let .row(.update(new, index)) = change,
+                       indexPath == IndexPath(row: index.row, section: index.section),
+                       let cell = view.cellForRow(at: indexPath) {
+                        let old = from[indexPath]
+                        cell.reconfigure(old: old, new: new)
+                        handledUpdates.append(index)
+                        break
```
Comment (enhorn) [inline Form/TableKit.swift:391]:
```suggestion
                           let old = from[indexPath]
                           cell.reconfigure(old: old, new: new)
                           handledUpdates.append(index)
                           break
```
Took a while before I realized the `if`-part was split into three rows. 😅

File: Form/TableKit.swift [hunk 3]
```diff
@@ -363,35 +363,60 @@ extension TableKit: TableAnimatable {
...
+                let row = table[tableIndex]
+                guard let index = from.firstIndex(where: { rowIdentifier(row) == rowIdentifier($0) }) else { continue }
 
-        changesCallbacker.callAll(with: changes)
-        callbacker.callAll(with: table)
+                if let cell = view.cellForRow(at: indexPath) {
+                    cell.updateBackground(forStyle: style, tableView: view, at: indexPath)
+
+                    let old = from[index]
+                    guard rowNeedsUpdate?(old, row) != false else {
```
Comment (enhorn) [inline Form/TableKit.swift:410]: Am I reading this right that the refresh of visible rows will be the done when `rowNeedsUpdate` is nil?
Perhaps we should describe that behaviour in the parameter comment.

Comment (nataliq-pp) [reply Form/TableKit.swift:410]: I think that's described already: `If provided, unnecessary reconfigure calls to visible rows could be avoided.` . So if not provided we will always assume the row needs update. But this part of the code is not changed, I just moved it to a function to understand the code better.

Comment (enhorn) [reply Form/TableKit.swift:410]: Alright, sounds good then.

**Review by enhorn — DISMISSED** [review]

File: Form/TableKit.swift (hunk 2 above)
Comment (nataliq-pp) [reply Form/TableKit.swift]: That's the indentation Xcode thinks we should have :D I changed it to `guard` now and added a missing `updateBackground` call. Please take another look @enhorn .

Comment (enhorn) [reply Form/TableKit.swift]: Ah, Xcode and it's weird indentation suggestions. 😅
Looks better with the guard. 💪

Comment (nataliq-pp) [reply Form/TableKit.swift]: Needs another +1 after the change 🙏

Comment (enhorn) [reply Form/TableKit.swift]: Oh, sorry. 😅

**Review by enhorn — APPROVED** [review]

**Review by enhorn — APPROVED** [review]

**Pull request merged**